src/models/ElasticDocument.php

```diff
@@ -89,11 +89,14 @@ protected function loadByEntry( craft\elements\Entry $entry )
 
     protected function loadByGlobalSet( craft\elements\GlobalSet $globalSet )
     {
-        $this->type = 'global';
+        // We'll prefix the type in order to avoid conflict with channels/structures
+        $this->type = 'global:' . $globalSet->handle;
```

Comment:
The type is used as part of the url to the document in elasticsearch, so should not contain ":"... maybe use dash instead?